src/analyses/apron/octApron.apron.ml

```diff
@@ -163,13 +163,6 @@ struct
     let open Queries in
     let d = ctx.local in
     match q with
-    | Assert e ->
-      begin match D.check_assert e ctx.local with
-        | `Top -> `Top
-        | `True -> `Lifted true
-        | `False -> `Lifted false
-        | _ -> `Bot
-      end
     | EvalInt e ->
       begin
         match D.get_int_val_for_cil_exp d e with
```

Comment:
This `EvalInt` can only evaluate (linear) expressions like `x - y`, but not constraints like `x < y`. So if there's an `assert(x < y);` and base queries the expression, this will just return `Top`, no?

I think this needs to distinguish the two cases and either evaluate an expression or check a constraint (via `check_assert` or whatever). How do the octApron tests even pass right now?

Reply:
I think I figured out why the tests even pass right now. It's because the few tests that we have are from the octagon analysis and have assertion in weird form like `assert(X - N == 0);`. The base analysis also queries the subexpression `X - N`, which octApron can evaluate as a linear expression, not a constraint, and returns the bounds `[0, 0]` for. And then the base analysis is doing the comparison between 0 and 0.

If it was `assert(X == N);` instead, then it wouldn't work, although it really should, because Apron can easily handle such constraint. It fails because `X == N` isn't handled (it's a constraint, not an expression) and `X` and `N` separately have no bounds, so base cannot compare any subexpressions either.